### DISCUSSION TITLE
GFX702 ROCm.

### DISCUSSION THREAD

**Discussion by mudapanda2** [ROCm/ROCm#1872] [General]
Hey there, I'm trying to hack and pack this together to work for my GPU and I'm growing frustrated with AMD at the moment.

I read the old thread regarding ROCM and Hawaii GPUs but I've got a question and a bone to pick I suppose.
The manual says such,

**The following list of GPUs are enabled in the ROCm software, though full support is not guaranteed:**
GFX7 GPUs,

It still says that as of Oct 11th 2022  https://docs.amd.com/bundle/ROCm-Release-Notes-v4.5/page/Hardware_and_Software_Support.html

I'm just really getting annoyed with GFX702 having been given the axe around 2.0, my specific 390 was released end of 2015, bought new in 2016, ROCM 2.0 hits around December 2018 and axes my card. That's two years on a new card before AMD deemed it deprecated and no longer supported. Why? Because it got lumped in with all the GPUs that AMD said it wasn't, that were 2013 or so released ergo it was too old now. Despite being two years old with the marketing.

 http://vrworld.com/2015/06/24/amd-moves-to-justify-radeon-300-series-rebrand/

See when we bought these cards we were told that Grenada XT and Grenada Pro weren't rebrands of Hawaii and stuff like that, only for two years down the road for them to be deprecated in the same group because they are "hawaii" Which was not what we were told when buying these bricks.

Now further barriers get created as HIP and such becomes a requirement and support for OpenCL is dropped in favor of it, referencing Blender here. So that's kind of injury to insult there. AMD has devs on Blenders roster so why keep OpenCL around? 

After swearing up down left and right that these cards were not simply Hawaii rebrands and then shoving them into the bucket when it was time for Hawaii to go out to pasture really sucks. They were $$$ for their day and in 2 years deemed obsolete, there's no way we can get some help for these cards? Been without GPU rendering in Blender for a year now because a requirement exists that for absolutely no good reason I can't meet because my "Not a rebrand" GPU ended up being a rebrand after all.

Is there no thought at all as to support for GFX702 or is it just a "Tough break pal, you got two years of basic function, who told you they weren't a rebrand?" situation. Hell only a few months later GFX803 came out, a worse card but at least support was kept for it until 4.5? 5.0? Who knows.  I'd straight up like to see some support for those of us who got duped by the Grenada Pro jazz. GFX702 and the other Grenada.

**Comment by xuhuisheng** [comment]
Not only gfx7 but also gfx8, wont get offical support from AMD.

Actually, only gfx9+ can get official support.  I guess the reason might be only gfx9 ISA can be support easily by DL.

And It is might be a hard journey for gfx7 to support ROCm, these cards loss some ISA likes permute, I try to compile ROCm for gfx702 target before, the math library cannot even compile successfully. That help me not to buy a gfx7 card to do more experiences.

Unfortunately, there is still some bugs on gfx803, even it still in the target list of math libraries. In my opinion there is little chance to support old card without AMD help, so we should help ourselves to just buy a gfx9+ card.

**Comment by mudapanda2** [comment]
Yeah but at least GFX8 got something like 5 years of support, up until 4.5
I bought this card being told it wasn't a Hawaii rebrand, It's Grenada, etc. Two years in, nope it's a hawaii now, and we don't support it. Thanks for spending top dollar on our best compute at the time. That's a real scammy scummy act on AMDs part and I don't think I'll follow you down the buying a gfx9 card, sorry.

They sold this claiming it wasn't what it really was, as a product built for whatever came, and it's been nothing but a lie. Two years into ownership and griping about losing ROCm support and get told "Ugh just buy a new card, hawaii is OOOOOLD" No, it wasn't old, it was a Grenada Pro according to AMD but it was easy to ignore supporting it and just call it Hawaii down the road. 

Patiently waiting hoping that support will come and there's never a move, it's been a real let down year and year with AMD and this card. Get lied to about what it was, cut off from development because of that truth, then sit on the bleachers and lose one after another applications that drop opencl for HIP. I'm sorry broski but AMDs conduct in this makes them evil as hell.

**Comment by xuhuisheng** [comment]
I don't think there is something like 5 years support for gfx8. Actually, ROCm drop offical support for gfx8 on ROCm-4.0 which released on 2020-12-15.

And even before they didn't test gfx8 on their CI environment, about ROCm-3.6 internal version, might be released in 2020-07.

The ROCm announced that Hawaii had been support about ROCm-1.9.3 which release 2018-12-13, but I had no gfx7 card cannot confirmed that.

But I know the timeline is useless for endpoint user, who just like me hoping ROCm will send resources to support OOOOOLD cards, Yes, I mean gfx8. I think the gfx9 offcial supporting is the limitation for ROCm team, just wish not drop gfx9 to dust too quick.
